lib/rules/no-classic-classes.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const { getImportModule } = require('../utils/utils');
+const { isObjectExpression } = require('../utils/types');
+
+const ERROR_MESSAGE_NO_CLASSIC_CLASSES =
+  'Native JS classes should be used instead of classic classes';
+
+function hasNoArguments(node) {
+  return node.arguments.length === 0;
+}
+
+function hasObjectArgument(node) {
+  return node.arguments.some(isObjectExpression);
+}
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+  ERROR_MESSAGE_NO_CLASSIC_CLASSES,
+
+  meta: {
+    docs: {
+      description: 'Disallow "classic" classes in favor of native JS classes',
+      category: 'Ember Octane',
+      recommended: false,
+    },
+    fixable: null, // or "code" or "whitespace"
+  },
+
+  create(context) {
+    function getModuleForNode(node) {
+      return getImportModule(context, node);
+    }
+
+    function reportNode(node) {
+      context.report(node, ERROR_MESSAGE_NO_CLASSIC_CLASSES);
+    }
+
+    return {
+      'CallExpression > MemberExpression[property.name="extend"]'(node) {
```

Comment:
Using a more specific selector here so that we only actually traverse into nodes that look like

```js
something.extend()
```

Otherwise we have to run the method here on _every_ `CallExpression` just to filter them down to instances that look like `something.extend()` manually.